### PR TITLE
Enable wasm builds on linux

### PR DESCRIPTION
**Pull request by cbouwense**
https://github.com/user-attachments/assets/385d8446-8e8d-4d73-805a-f3ac41b81bb7

The diff is so huge because of adding in the recompiled Odin that is configured for wasm exports. You can basically just ignore `libs/`, that's where that is.

I should add documentation what to actually do / an environment bootstrap script, since you need the emscripten SDK and of course odin, and then all of their dependencies. Honestly, might be nice to just have a directory in the repo with those.

### CODE REVIEW

**Comment by cbouwense** [comment]
It might be good for me to also add in a build check for this

File: web/main.c [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#include <emscripten/emscripten.h>
```
Comment (cbouwense) [inline web/main.c:1]: these files in `web/` are all copypasta from that template repo, no need to review

File: libs/raylib/LICENSE [hunk 2]
```diff
@@ -0,0 +1,16 @@
+Copyright (c) 2013-2023 Ramon Santamaria (@raysan5)
```
Comment (cbouwense) [inline libs/raylib/LICENSE:1]: everything in lib is copypasta from that template repo, no need to review

File: game/game.odin
Comment (ckuras) [inline game/game.odin:1]: Love the refactor

**Comment by ckuras** [comment]
So everything in libs/ is to support building for all of those platforms?

**Comment by cbouwense** [comment]
> So everything in libs/ is to support building for all of those platforms?

@ckuras yeah exactly

File: game/game.odin
Comment (cbouwense) [reply game/game.odin:1]: Thanks, it looks nice, but it's actually necessary so main.odin can import those procs and do it's wasm magic stuff with emscripten I don't understand

**Comment by ckuras** [comment]
@cbouwense got it. It would be cool to move that all to the docker image that runs in the action, but this is good to get us up and running. Maybe if I make it a standalone action I can figure that out.

**Review by ckuras — APPROVED** [review]
:^)

**Pull request merged**